### PR TITLE
chore: enable kres

### PR DESCRIPTION
**Pull request by frezbo**
Enable kres and update repo branch settings.

Notable changes:

* `Makefile`, `Dockerfile`, Github Actions workflow etc are autogenerated, run a `make rekres` to update things.
* Project config is handled by `.kres.yaml`, additional options can be added and a `make rekres` should update all.
* There are now more checks for commit messages. (can be configured via `.kres.yaml`)

Pull requests require status checks and approval.

Usually a PR can be merged by adding a `/m` comment. We do this in the org repos to avoid merge commits that has no trusted GPG signature (GitHub merge commits doesn;t do a `ff` merge and adds their own GPG signature)

I have disabled the GPG signature to be enforced for a commit.

Fixes: #14

### CODE REVIEW

File: Dockerfile
Comment (robinelfrink) [inline Dockerfile:19]: Doesn't this remove the ability to use this container as system extension?

Comment (frezbo) [reply Dockerfile:19]: Ahh I didn't realize this was supposed to be used like that, I though this would go as an official system extension? :thinking:

Comment (robinelfrink) [reply Dockerfile:19]: I'm not sure, let's ask @jonkerj what was agreed when this code was moved to `siderolabs/`.

Comment (frezbo) [reply Dockerfile:19]: cool, else I can add this back, kres should support it

Comment (robinelfrink) [reply Dockerfile:19]: Okay, thanks. Can you leave the last lines in as well, including `ENTRYPOINT`? This way the container is usable as a `Pod` as well as a system extension.

I believe the idea was have this repository for the extension short term at least, and _maybe_ add it to siderolabs/extensions later.

**Comment by JinxCappa** [comment]
To all parties involved,

I would like to make a strong case to have this tool added to the official extensions repo. While standard Talos users can continue to use this tool the original way designed, having this be an extension simplifies implementation as extensions can be baked into the image so that now all vSphere VMs can enable this right from clone/deploy. The simplest and easiest way to do that I believe is using Image Factory. 

The biggest reason honestly to add this as an official extension is for Omni I think. Being able to create an ova from within Omni with the vmtoolsd extension already added would be great. An even bigger reason is for upgradability. While now Omni connectivity settings can be added to custom Talos builds with custom extensions, not unifying these two repos breaks Omni's design of managing and upgrading these VMs as Omni does not allow for incorporation of non official extensions into images built from within it's interface. At least I believe so; who knows, could be wrong.

So again, I hope my reasons sound well.. reasonable. I actually do have a commit burning a hole in my fork of the extensions repo begging for a pull request but I digress and will wait for the maintainers of this repo to come to a decision.

**Comment by robinelfrink** [comment]
Hi @JinxCappa,

I agree having `talos-vmtoolsd` in the official extensions repository would be great. I don't see any reason why that couldn't be done, even if this repository builds it's own extension-container.

Note however that this repository is a community effort, and none of the maintainers are affiliated with Sidero. They just offered to host the repository under `siderolabs/'.

So my advice would be to just create that PR, parallel to what we are doing here.

File: Dockerfile
Comment (jonkerj) [reply Dockerfile:19]: > I'm not sure, let's ask @jonkerj what was agreed when this code was moved to `siderolabs/`.

Yes, the ambition was to have `talos-vmtoolsd` as an official system extension at some point. No promises were made, but we'd like to put in the effort to get there.

**Comment by JinxCappa** [comment]
I appreciate the response and am looking forward to submitting my PR.

As a question to @frezbo, as this repo stands now the releases are not tagged with 'v{semver}' but just the plain version instead. With the addition of kres to this repo, will this be changed to match the release format of siderolabs repos? I ask as this will affect pulling the source to build the extension as well as the renovate regex comment in the PR. View the below links for reference:

[pkgs.yaml](https://github.com/JinxCappa/extensions/blob/93c1d6cd6782ce56c4065a05a93b6d1e9cb56d0f/guest-agents/talos-vmtoolsd/pkg.yaml#L13)
[vars.yaml](https://github.com/JinxCappa/extensions/blob/93c1d6cd6782ce56c4065a05a93b6d1e9cb56d0f/guest-agents/vars.yaml#L9)

If so, I don't have an issue submitting my PR to extensions but note that it will have to be modified if this PR gets finalized and reformats release tags.

**Comment by frezbo** [comment]
> I appreciate the response and am looking forward to submitting my PR.
> 
> As a question to @frezbo, as this repo stands now the releases are not tagged with 'v{semver}' but just the plain version instead. With the addition of kres to this repo, will this be changed to match the release format of siderolabs repos? I ask as this will affect pulling the source to build the extension as well as the renovate regex comment in the PR. View the below links for reference:
> 
> [pkgs.yaml](https://github.com/JinxCappa/extensions/blob/93c1d6cd6782ce56c4065a05a93b6d1e9cb56d0f/guest-agents/talos-vmtoolsd/pkg.yaml#L13) [vars.yaml](https://github.com/JinxCappa/extensions/blob/93c1d6cd6782ce56c4065a05a93b6d1e9cb56d0f/guest-agents/vars.yaml#L9)
> 
> If so, I don't have an issue submitting my PR to extensions but note that it will have to be modified if this PR gets finalized and reformats release tags.

Yes, it would starting with `v`, at least the CI expects that standard for tags

File: Dockerfile
Comment (frezbo) [reply Dockerfile:19]: Okay, I've resolved this

**Comment by robinelfrink** [comment]
@frezbo thanks! Container generated using the modified Makefile works as DaemonSet as well as Talos System Extension.

Judging by the new github workflow code, we would still need to manually create tags, right?

**Comment by frezbo** [comment]
> Judging by the new github workflow code, we would still need to manually create tags, right?

yes, just make sure it has the `v` prefix

**Comment by frezbo** [comment]
> @frezbo thanks! Container generated using the modified Makefile works as DaemonSet as well as Talos System Extension.
> 
> Judging by the new github workflow code, we would still need to manually create tags, right?

if you could +1 (approval), I can then get it merged

**Review by robinelfrink — APPROVED** [review]

**Pull request merged**
